cli/bin/grain.js

```diff
@@ -71,144 +79,165 @@ const optionApplicator = (Option) =>
-  )
-  .profileOption(
-    "--release",
-    "compile using the release profile (production mode)"
-  )
-  .forwardOption(
-    "--experimental-wasm-tail-call",
-    "enables tail-call optimization"
-  )
-  .forwardOption("--debug", "compile with debugging information")
-  .forwardOption("--wat", "additionally produce a WebAssembly Text (.wat) file")
-  .forwardOption(
-    "--hide-locs",
-    "hide locations from intermediate trees. Only has an effect with `--verbose`"
-  )
-  .forwardOption("--lsp", "generate lsp errors and warnings only")
-  .forwardOption("--no-color", "disable colored output")
-  .forwardOption("--no-gc", "turn off reference counting garbage collection")
-  .forwardOption(
-    "--no-bulk-memory",
-    "polyfill WebAssembly bulk memory instructions"
-  )
-  .forwardOption(
-    "--wasi-polyfill <filename>",
-    "path to custom WASI implementation"
-  )
-  .forwardOption(
-    "--use-start-section",
-    "replaces the _start export with a start section during linking"
-  )
-  .forwardOption("--no-link", "disable static linking")
-  .forwardOption(
-    "--no-pervasives",
-    "don't automatically import the Grain Pervasives module"
-  )
+  .addOption(new commander.Option("-p, --print-output").hideHelp())
   .forwardOption("-o <filename>", "output filename")
-  .forwardOption("-O <level>", "set the optimization level")
-  .forwardOption(
-    "--parser-debug-level <level>",
-    "debugging level for parser output"
-  )
-  .forwardOption(
-    "--memory-base <addr>",
-    "set the base address for the Grain heap"
-  )
-  .forwardOption("--source-map", "generate source maps")
-  .forwardOption("--strict-sequence", "enable strict sequencing")
-  .forwardOption(
-    "--verbose",
-    "print critical information at various stages of compilation"
-  )
-  // The root command that compiles & runs
-  .arguments("<file>")
   .action(function (file, options, program) {
-    exec.grainc(file, program);
+    exec.grainc(file, options, program);
     if (options.o) {
-      run(options.o, program.opts());
+      run(options.o, options);
     } else {
-      run(file.replace(/\.gr$/, ".gr.wasm"), program.opts());
+      run(file.replace(/\.gr$/, ".gr.wasm"), options);
     }
   });
 
 program
   .command("compile <file>")
   .description("compile a grain program into wasm")
-  .action(function (file) {
-    // The compile subcommand inherits all behaviors/options of the
-    // top level grain command
-    exec.grainc(file, program);
-  });
+  .forwardOption("-o <filename>", "output filename")
+  .action(exec.grainc);
 
 program
   .command("run <file>")
   .description("run a wasm file with grain's javascript runner")
-  .action(function (wasmFile) {
-    // The run subcommand inherits all options of the
-    // top level grain command
-    run(wasmFile, program.opts());
-  });
+  .addOption(new commander.Option("-p, --print-output").hideHelp())
+  .action(run);
 
 program
   .command("lsp <file>")
```

Comment:
Will need to remember to update this for the new LSP

Reply:
Yeah, I've been doing the merge and resolves